server/services/activitiesService.ts

```diff
@@ -243,6 +249,83 @@ export default class ActivitiesService {
     return this.activitiesApiClient.getAppointment(appointmentId, user)
   }
 
+  async getAppointmentDetails(appointmentId: number, user: ServiceUser): Promise<AppointmentDetails> {
+    const appointment = await this.getAppointment(appointmentId, user)
+
+    const locationsMap = (
+      await this.prisonApiClient.getLocationsForEventType(appointment.prisonCode, 'APP', user)
+    ).reduce((map, location) => {
+      return map.set(location.locationId, location)
+    }, new Map<number, LocationLenient>())
+
+    const occurrenceSummaries = appointment.occurrences.map(occurrence => {
+      return {
+        id: occurrence.id,
+        internalLocation: locationsMap.get(occurrence.internalLocationId),
+        inCell: occurrence.inCell,
+        startDate: new Date(occurrence.startDate),
+        startTime: new Date(`${occurrence.startDate}T${occurrence.startTime}:00`),
+        endTime: occurrence.endTime !== null ? new Date(`${occurrence.startDate}T${occurrence.endTime}:00`) : null,
+        comment: occurrence.comment,
+        isCancelled: false,
+        updated: null,
+        updatedBy: null,
+        isEdited: false,
+      } as AppointmentOccurrenceSummary
+    })
+
+    const prisonerNumbers = appointment.occurrences
+      .map(occurrence =>
+        occurrence.allocations.map(allocation => {
+          return allocation.prisonerNumber
+        }),
+      )
+      .flat()
+    const prisoners = await this.prisonerSearchApiClient.searchByPrisonerNumbers({ prisonerNumbers }, user)
+    const prisonerMap = prisoners.reduce((map, prisoner) => {
+      return map.set(prisoner.prisonerNumber, prisoner)
+    }, new Map<string, Prisoner>())
+
+    let createdBy = null
+    try {
+      const createdByUserDetail = await this.prisonApiClient.getUserByUsername(appointment.createdBy, user)
+      createdBy = `${createdByUserDetail.firstName} ${createdByUserDetail.lastName}`
+    } catch (e) {
+      // If the username isn't found, log the error and continue
+      if (e.status === 404) {
+        logger.info(`Couldn't get user with username, ${appointment.createdBy}`)
+      } else {
+        throw e
+      }
+    }
+
+    let updatedBy = null
+    if (appointment.createdBy === appointment.updatedBy) {
+      updatedBy = createdBy
+    } else if (appointment.updatedBy !== null) {
+      const updatedByUserDetail = await this.prisonApiClient.getUserByUsername(appointment.updatedBy, user)
```

Comment:
I've updated the logic to catch the 404 for both calls. I've also improved the logic to run both api calls in parallel (where both are needed).

While making that change I also thought it made more sense to return the full user object in the appointment details to improve extensibility, rather than pulling out the first name and last name at the service level.

I'm now passing the user object directly down to the nunjucks template and pulling out the name with a new `fullName` filter, this is then passed to the `initialiseName` filter to render the user's name. The logic for this being that the rendering of the user details should be handled by the view (in my opinion anyway).